device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/https/HttpsIotHubConnection.java

```diff
@@ -80,7 +80,29 @@ public IotHubStatusCode sendMessage(Message message) throws TransportException
     {
         synchronized (HTTPS_CONNECTION_LOCK)
         {
-            HttpsMessage httpsMessage = HttpsSingleMessage.parseHttpsMessage(message);
+            // Here we check if it's a bulk message and serialize it.
+            HttpsMessage httpsMessage = null;
+
+            if (message.isBulk())
+            {
+                HttpsBatchMessage batchMessage = new HttpsBatchMessage();
+                for (Message singleMessage : message.getNestedMessages())
+                {
+                    try
+                    {
+                        batchMessage.addMessage(HttpsSingleMessage.parseHttpsMessage(singleMessage));
+                    }
+                    catch (Exception e)
+                    {
+                        throw new TransportException(e);
```

Comment:
```throw new TransportException("Failed to create http batch message", e);``` to add context

Reply:
the inner exception already has this message. I will add this on top of it for more clarity